libs/commands/publish/__fixtures__/relative-file-specs/packages/package-d/package.json

```diff
@@ -0,0 +1,7 @@
+{
+  "name": "package-d",
+  "version": "1.0.0",
+  "peerDependencies": {
+    "package-1": "workspace:*2.3.4"
```

Comment:
I have never seen this syntax before (a `*` combined with a version number), do you have a reference for it please? There was nothing on https://pnpm.io/workspaces#workspace-protocol-workspace based on a quick look

Reply:
James, 

Indeed, I have not seen the * combined with a version number syntax before either. Here's my logic, and IMHO you are free to disagree with it because I think there is a certain arbitrariness about this whole issue.

Feel free to skip to the TLDR... 

I have yet to see a specification that really pins down both what is allowed for the workspace syntax source and the computed package output. In all cases, I find only examples, not specification, and these are generally so incomplete as to allow ambiguity. This is certainly true of the pnpm hyperlink that you quoted. Looking at the reference that describes what we are trying to compute, https://docs.npmjs.com/about-semantic-versioning, there is also an ambiguity. You could argue that the authors of that page intended to disallow *major.minor.patch as a syntax by its omission. Sure, but there two (and maybe three) arguments against that. First, it is a reasonable extrapolation of the examples that *1.0.4 should be allowed. Second, the meaning of *major.minor.patch could be distinct from the meaning of * or x since it establishes a minimum version whereas the latter two do not. Finally (third), the "npm SemVer calculator" linked on that page, https://semver.npmjs.com/#syntax-examples, actually allows the *major.minor.patch syntax although it doesn't produce the sensible result.

TLDR:

While I think the use of *major.minor.patch syntax by a user is unlikely, there is currently no SemVer syntax validation in Lerna to exclude it. I think that maybe there should be but felt it was beyond the scope of this change to introduce a semantic versioning validator and kick out exceptions (besides being beyond my familiarity with Lerna code). Instead, I decided that I wanted to ensure that such syntax would produce sensible, unsurprising behavior. I certainly didn't want the syntax to be converted to '^', '~' or something else because of an internal code change such as falling through a switch statement. So I added the test to attempt to circumvent future surprising behavior.

Reply:
Let's please remove it, we shouldn't support unexpected values, folks will have bigger problems with their package managers if they use them and they shouldn't expect them to work in lerna either.

FYI `1.0.0*`, `1*.0.0`, `1.0*.0` also doesn't blow up that online filter (https://semver.npmjs.com/#syntax-examples), but that doesn't mean we should support them either.

The semver spec covers the `MAJOR.MINOR.PATCH` etc, the package manager tooling covers the usage of ranges, and clearly this isn't a supported case for the package managers so shouldn't be for lerna either.

Reply:
James, 

Please clarify whether you would like to a) remove the test, and/or b) prevent the behavior. There are probably numerous tests that I could add that demonstrate quirky behavior prior to this enhancement, and these cases could only be removed by introducing syntactic validation of the "workspace:" string.